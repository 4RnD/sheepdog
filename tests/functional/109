#!/bin/bash

# Test sheep -z/--zone

. ./common

err=0

function testStartZoneSucceeded
{
	zone="$1"
	mkdir -p "$STORE/0"
	if $SHEEP "$STORE/0" -z "$zone" ; then
		_wait_for_sheep 1
		if grep daemon "$STORE/0/sheep.log" >/dev/null 2>&1 ; then
			$DOG node list
		else
			err=1
		fi
	else
		err=1
	fi
	_cleanup
}

function testStartZoneFailed
{
	zone="$1"
	mkdir -p "$STORE/0"
	$SHEEP "$STORE/0" -z "$zone" && err=1
	_cleanup
}

testStartZoneSucceeded 0
testStartZoneSucceeded 1
testStartZoneSucceeded 4294967295 # UINT32_MAX
testStartZoneFailed 4294967296 # UINT32_MAX + 1
testStartZoneFailed 9223372036854775808 # INT64_MAX + 1
testStartZoneFailed -1
testStartZoneFailed a
testStartZoneFailed 42a
testStartZoneFailed +
testStartZoneFailed -

exit $err
