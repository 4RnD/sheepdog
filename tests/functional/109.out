QA output created by 109
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          1
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128 4294967295
Invalid zone id '4294967296': must be an integer between 0 and 4294967295
Invalid zone id '9223372036854775808': must be an integer between 0 and 4294967295
Invalid zone id '-1': must be an integer between 0 and 4294967295
Invalid zone id 'a': must be an integer between 0 and 4294967295
Invalid zone id '42a': must be an integer between 0 and 4294967295
Invalid zone id '+': must be an integer between 0 and 4294967295
Invalid zone id '-': must be an integer between 0 and 4294967295
